osu.Game.Rulesets.Osu/Objects/OsuHitObject.cs

```diff
@@ -130,5 +130,7 @@ protected override void ApplyDefaultsToSelf(ControlPointInfo controlPointInfo, B
         }
 
         protected override HitWindows CreateHitWindows() => new OsuHitWindows();
+
+        public override string ToEditorString() => (IndexInCurrentCombo + 1).ToString();
```

Comment:
Not sure about doing this locally, either. There should be avenues available to do this in a generic fashion (`IHasCombo` and `IHasColumn`, I believe).

Reply:
Not sure about that either, because users should be able to implement a full ruleset without being required to use `IHasCombo` and `IHasColumn`.

I think best is to put this inside `HitObjectComposer`, as a virtual `ConvertHitObjectToString()` or `ConvertSelectionToString()` (if the concatenation is moved inside) method. By default, sure, use `IHasCombo` and `IHasColumn`.